### PR TITLE
Backtrace information is only resolved with feature flag

### PR DESCRIPTION
**Pull request by AshantiMutinta**
## PR summary
-Added a feature flag for backtracing during sync
## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

- [ ] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [ ] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

File: crates/locksmith/src/tracker.rs [hunk 1]
```diff
@@ -87,6 +92,20 @@ Backtrace at the moment of guard creation follows:
...
+        !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        !!! IMMORTAL LOCK GUARD FOUND !!!
+        !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+{type:?} guard {puid} lived for > {time} seconds.{annotation}",
+            type=self.lock_type,
+            puid=self.puid,
+            time=IMMORTAL_TIMEOUT.as_secs(),
+            annotation=annotation
+        );
```
Comment (struktured) [inline crates/locksmith/src/tracker.rs]: What is purpose of this new `error!` block here?

Comment (AshantiMutinta) [reply crates/locksmith/src/tracker.rs]: Conditionally compiled, wanted to provide an error block that doesn't mention the backtrace and just having two blocks of code didn't seem too bad the macro + flag could catch them but could always clean this up

File: crates/locksmith/Cargo.toml [hunk 2]
```diff
@@ -18,3 +18,13 @@ lazy_static = "=1.2.0"
 log = "=0.4.8"
 parking_lot ="=0.9.0"
 snowflake = "=1.3.0"
+
+[features]
+#default brings in no features
+default =[]
+
+#this feature allows the compilation with backtraces involved
+sync_backtrace = []
```
Comment (neonphog) [inline crates/locksmith/Cargo.toml]: in tracker.rs, it's `sync_backtrace_capture`

Comment (AshantiMutinta) [reply crates/locksmith/Cargo.toml]: good catch error during renaming :)

**Comment by AshantiMutinta** [comment]
State - not going to be used so closing this

**Comment by maackle** [comment]
I think we should give this a try!

One question @AshantiMutinta why is the feature called `sync_backtrace`? Can we just call it `backtrace` or something?

**Comment by AshantiMutinta** [comment]
> 
> 
> I think we should give this a try!
> 
> One question @AshantiMutinta why is the feature called `sync_backtrace`? Can we just call it `backtrace` or something?

awesome! I think we should call it backtrace I was hoping maybe even "Debug Mode" but backtrace is much better

**Comment by maackle** [comment]
> I feel like we will need to make sure our CI exercises both versions, something like:

@neonphog I feel like CI only needs to exercise the one we'll be using in production (default), since the backtraces one is one we'd only be using for diagnosis, and that is the case already. Thoughts?

**Comment by neonphog** [comment]
@maackle 

> I feel like CI only needs to exercise the one we'll be using in production (default), since the backtraces one is one we'd only be using for diagnosis, and that is the case already. Thoughts?

Yeah, for a debug feature that makes sense to me :+1:

**Pull request merged**